pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/ProtoParserTools.java

```diff
@@ -227,11 +224,15 @@ public static String readString(final ReadableSequentialData input) throws IOExc
      * @param input the input to read from
      * @return read Bytes object, this can be a copy or a direct reference to inputs data. So it has same life span
      * of InputData
-     * @throws IOException If there was a problem reading
      */
-    public static Bytes readBytes(final ReadableSequentialData input) throws IOException {
+    public static Bytes readBytes(final ReadableSequentialData input) {
         final int length = input.readVarInt(false);
-        return input.readBytes(length);
+        validateInput(input, length);
```

Comment:
Is it really worth a method call for 2 copies of one if? Or is it for testing? If for testing then maybe makes sense and add java doc to validate method

Reply:
Initially this method was intended to be used in multiple places but it turned out to be unnecessary. So, I inlined it as you suggested.